src/transformState.js

```diff
@@ -5,7 +5,29 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const currentObject of actions) {
+    switch (currentObject.type) {
+      case 'addProperties':
+        Object.assign(state, currentObject.extraData);
+        break;
+      case 'removeProperties':
+        currentObject.keysToRemove.forEach(elem => delete state[elem]);
+        break;
+
+      case 'clear':
+        toClearState(state);
+        break;
+      default:
+        return null;
+    }
+  }
+
+  return state;
 }
 
+function toClearState(state) {
+  for (const elem in state) {
```

Comment:
`elem` :harold:
![image](https://user-images.githubusercontent.com/75496546/225561607-aaa9b7ef-d424-4c6a-a048-9320d6fb3d13.png)

Reply:
it was my favorite cigarette)))